name: Run Tests

on:
  workflow_run:
    workflows: ["Pull Request Open"]
    types: [completed]

concurrency:
  group: uat-${{ github.event.workflow_run.pull_requests[0].number }}
  cancel-in-progress: true

jobs:
  test-fe:
    name: Test Frontend (${{ github.event.workflow_run.pull_requests[0].number }})
    secrets: inherit
    uses: ./.github/workflows/.nodejs-test.yml
    with:
      environment: ${{ github.event.workflow_run.pull_requests[0].number }}

  test-be:
    name: Test Backend (${{ github.event.workflow_run.pull_requests[0].number }})
    secrets: inherit
    uses: ./.github/workflows/.java-test.yml
    with:
      environment: ${{ github.event.workflow_run.pull_requests[0].number }}

  results:
    name: Test Results
    needs: [test-fe, test-be]
    if: always()
    runs-on: ubuntu-24.04
    steps:
      - if: contains(needs.*.result, 'failure')||contains(needs.*.result, 'canceled')
        run: echo "At least one job has failed." && exit 1
      - run: echo "Success!"